### PR TITLE
Allow to search for implicits in macros

### PR DESCRIPTION
**Pull request by nicolasstucki**

### CODE REVIEW

File: library/src-3.x/scala/quoted/matching/package.scala [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  /** Find an implicit of type `T` in the current scope given by `qctx`.
+   *  Return `Some` containing the expression of the implicit or
+   * `None` if implicit resolution failed.
+   *
+   *  @tparam T type of the implicit parameter
+   *  @param tpe quoted type of the implicit parameter
+   *  @param qctx current context
+   */
+  def getImplicitExpr[T] given (tpe: Type[T], qctx: QuoteContext): Option[Expr[T]] = {
```
[the remark this answers is not in the document]
Comment (nicolasstucki) [reply library/src-3.x/scala/quoted/matching/package.scala]: Done

File: library/src/scala/quoted/matching/package.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  /** Find an implicit of type `T` in the current scope given by `qctx`.
+   *  Return `Some` containing the expression of the implicit or
+   * `None` if implicit resolution failed.
+   *
+   *  @tparam T type of the implicit parameter
+   *  @param tpe quoted type of the implicit parameter
+   *  @param qctx current context
+   */
+  def searchImplicitExpr[T] given (tpe: Type[T], qctx: QuoteContext): Option[Expr[T]] = {
```
Comment (liufengyun) [inline library/src/scala/quoted/matching/package.scala]: One minor confusion with the name `searchImplicitExpr`: on a first look, it seems to mean to look for implicits of the type `Expr[T]`.

Comment (nicolasstucki) [reply library/src/scala/quoted/matching/package.scala]: How else could I call it? I could also use `searchImplicit` if the `Reflection.searchImplicit` is renamed. Then what should I call that one? Could be `searchImplicitArg`

Comment (liufengyun) [reply library/src/scala/quoted/matching/package.scala]: Can they both be called `searchImplicit`?

**Pull request merged**